tensorflow_addons/optimizers/tests/lazy_adam_test.py

```diff
@@ -252,59 +254,57 @@ def testTensorLearningRate(self):
                     self.assertAllCloseAccordingToType(var0_np, self.evaluate(var0))
                     self.assertAllCloseAccordingToType(var1_np, self.evaluate(var1))
 
-    @test_utils.run_deprecated_v1
-    def testSharing(self):
-        for dtype in [tf.dtypes.half, tf.dtypes.float32, tf.dtypes.float64]:
-            with self.cached_session():
-                # Initialize tf for numpy implementation.
-                m0, v0, m1, v1 = 0.0, 0.0, 0.0, 0.0
-                var0_np = np.array([1.0, 2.0], dtype=dtype.as_numpy_dtype)
-                grads0_np = np.array([0.1, 0.1], dtype=dtype.as_numpy_dtype)
-                var1_np = np.array([3.0, 4.0], dtype=dtype.as_numpy_dtype)
-                grads1_np = np.array([0.01, 0.01], dtype=dtype.as_numpy_dtype)
-
-                var0 = tf.Variable(var0_np)
-                var1 = tf.Variable(var1_np)
-                grads0 = tf.constant(grads0_np)
-                grads1 = tf.constant(grads1_np)
-                opt = lazy_adam.LazyAdam()
-                update1 = opt.apply_gradients(zip([grads0, grads1], [var0, var1]))
-                update2 = opt.apply_gradients(zip([grads0, grads1], [var0, var1]))
-                self.evaluate(tf.compat.v1.global_variables_initializer())
-
-                beta_1_power, beta_2_power = get_beta_accumulators(opt, dtype)
 
-                # Fetch params to validate initial values
-                self.assertAllClose([1.0, 2.0], self.evaluate(var0))
-                self.assertAllClose([3.0, 4.0], self.evaluate(var1))
+@pytest.mark.usefixtures("maybe_run_functions_eagerly")
+@pytest.mark.parametrize("dtype", [tf.half, tf.float32, tf.float64])
+def test_sharing(dtype):
+    # Initialize tf for numpy implementation.
+    m0, v0, m1, v1 = 0.0, 0.0, 0.0, 0.0
+    var0_np = np.array([1.0, 2.0], dtype=dtype.as_numpy_dtype)
+    grads0_np = np.array([0.1, 0.1], dtype=dtype.as_numpy_dtype)
+    var1_np = np.array([3.0, 4.0], dtype=dtype.as_numpy_dtype)
+    grads1_np = np.array([0.01, 0.01], dtype=dtype.as_numpy_dtype)
+
+    var0 = tf.Variable(var0_np)
+    var1 = tf.Variable(var1_np)
+    grads0 = tf.constant(grads0_np)
+    grads1 = tf.constant(grads1_np)
+    opt = lazy_adam.LazyAdam()
+    _ = opt.apply_gradients(zip([grads0, grads1], [var0, var1]))
+
+    beta_1_power, beta_2_power = get_beta_accumulators(opt, dtype)
+
+    # Fetch params to validate initial values
+    np.allclose(np.asanyarray([1.0, 2.0]), var0.numpy())
+    np.allclose(np.asanyarray([3.0, 4.0]), var1.numpy())
+
+    # Run 3 steps of intertwined Adam1 and Adam2.
+    for t in range(3):
```

Comment:
One thing is strange, in this loops, the TF tensors are never updated. Before there was a `run()` per iteration. That's very strange. @ssaishruthi we're trying to remove the `@test_utils.run_deprecated_v1` and run the test in eager mode, could you help us on this one?